app/helpers/questions.py

```diff
@@ -0,0 +1,77 @@
+import yaml
+import inflection
+import re
+import os
+
+from collections import OrderedDict
+
+
+class QuestionsLoader(object):
+
+    def __init__(self,
+                 manifest='questions_manifest.yml',
+                 questions_dir=''):
+
+        with open(manifest, "r") as file:
+            section_order = yaml.load(file)
+
+        self._directory = questions_dir
+        self._question_cache = {}
+        self.sections = [
+            self.__populate_section__(s) for s in section_order
+        ]
+
+    def get_section(self, requested_section):
+
+        for section in self.sections:
+            if section["id"] == requested_section:
+                return section
+
+    def get_question(self, question):
+
+        if question not in self._question_cache:
+
+            question_file = self._directory + question + ".yml"
+
+            if not os.path.isfile(question_file):
+                self._question_cache[question] = {}
+                return {}
+
+            with open(question_file, "r") as file:
+                question_content = yaml.load(file)
+
+            question_content["id"] = question
+
+            # wrong way to do it? question should be shown by default.
+            question_content["depends_on_lots"] = (
+                self.__get_dependent_lots__(question_content["dependsOnLots"])
+            ) if "dependsOnLots" in question_content else (
+                ["saas", "paas", "iaas", "scs"]
+            )
+
+            self._question_cache[question] = question_content
+
+        return self._question_cache[question]
+
+    def __populate_section__(self, section):
```

Comment:
This style of method name should be reserved for Python [special methods](https://docs.python.org/2/reference/datamodel.html#special-method-names).